src/pages/MonthlyGood/MonthlyGoodSetupPage.vue

```diff
@@ -393,27 +426,107 @@ export default {
+		preFetch(config, client, { route }) {
+			// When SUBS-609 login exp ends, this route can be set back to
+			// meta: {
+			// 	activeLoginRequired: true,
+			// }
+			// in routes.js
+
+			// Get exp value for SUBS-609
+			return client.query({
+				query: expQuery
+			})
+				.then(() => client.query({
+					query: experimentAssignmentQuery, variables: { id: 'mg_login_after_setup' }
+				}))
+				.then(({ data }) => {
+					const loginAfterSetupExpVersion = data?.experiment?.version ?? {};
+					// Control version
+					// Auth status should be checked.
+					if (loginAfterSetupExpVersion === 'control') {
+						return client.query({
+							query: authenticationQuery,
+							// eslint-disable-next-line no-shadow
+						}).then(({ data }) => {
+							if (!data.my) {
+								throw new Error('api.authenticationRequired');
+							}
+							// Route requires active login
+							if (!checkLastLoginTime(data, 'activeLoginDuration', 3600)) {
+								throw new Error('activeLoginRequired');
+							}
+						}).catch(() => {
+							// Auth error will be caught here, redirect to login.
+							return Promise.reject({
+								path: '/ui-login',
+								query: { doneUrl: route.fullPath }
+							});
+						});
+					}
+					// Shown version
+					// Auth status should not be checked, continue with pageQuery
+				})
+				.then(() => client.query({ query: pageQuery }));
 		},
 	},
 	created() {
+		// Track experiment
+		try {
+			const expQueryResult = this.apollo.readQuery({
+				query: experimentAssignmentQuery, variables: { id: 'mg_login_after_setup' }
+			});
+			const loginAfterSetupExpVersion = expQueryResult?.experiment?.version ?? {};
+			if (loginAfterSetupExpVersion === 'control') {
+				this.$kvTrackEvent('MonthlyGood', 'EXP-SUBS-609-Jan2021', 'a');
+			} else if (loginAfterSetupExpVersion === 'shown') {
+				this.$kvTrackEvent('MonthlyGood', 'EXP-SUBS-609-Jan2021', 'b');
+			}
+		} catch (e) {
+			logReadQueryError(e, 'MonthlyGoodSetupPage expQueryResult');
+		}
+
+		try {
+			const pageQueryResult = this.apollo.readQuery({
+				query: pageQuery,
+			});
+			this.isMGTaglineActive = pageQueryResult?.general?.mgDonationTaglineActive?.value === 'true' || false;
+			this.isMonthlyGoodSubscriber = pageQueryResult?.my?.autoDeposit?.isSubscriber ?? false;
+			this.hasAutoDeposits = pageQueryResult?.my?.autoDeposit ?? false;
+			this.hasAutoLending = pageQueryResult?.my?.autolendProfile?.isEnabled ?? false;
+			this.legacySubs = pageQueryResult?.my?.subscriptions?.values ?? [];
+			this.hasLegacySubscription = this.legacySubs.length > 0;
+			this.hasActiveLogin = !!pageQueryResult?.my;
+		} catch (e) {
+			logReadQueryError(e, 'MonthlyGoodSetupPage pageQuery');
+		}
+
 		// Sanitize and set initial form values.
+		// Initial group from prop
 		if (this.lendingCategories.find(category => category.value === this.category)) {
 			this.selectedGroup = this.category;
 		}
 
-		if (!Number.isNaN(Number(this.amount))) {
```

Comment:
Removing the `Number()` call means that things like `undefined` or `'blabla'` will be allowed, since those are not `NaN`. `isNaN` returns true only for values that are `NaN` and are type `Number` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#examples)

Reply:
I believe this call is redundant. The prop is coming from the route, and in the route definition:
`amount: Number(route.query.amount)`

src/router/routes.js:105

Reply:
Aha, yeah I think that should take care of that!